File: jenjin-core/src/main/java/com/binarymonks/jj/render/nodes/TextureProvider.java
package com.binarymonks.jj.render.nodes;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.TextureRegion;

public interface TextureProvider {

    TextureRegion getFrame(float rotationD);

    void dispose();

    public static class SimpleTextureProvider implements TextureProvider {

        Texture texture;
        TextureRegion renderFrame;

        public SimpleTextureProvider(Texture texture) {
            this.texture = texture;
            this.renderFrame = new TextureRegion(texture);
        }

        @Override
        public TextureRegion getFrame(float rotationD) {
            return renderFrame;
        }

        @Override
        public void dispose() {
            texture.dispose();
        }
    }
}

File: jenjin-core/src/main/kotlin/com/binarymonks/jj/core/render/nodes/TextureProvider.kt
package com.binarymonks.jj.core.render.nodes

import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.TextureRegion

interface TextureProvider {

    fun getFrame(rotationD: Float): TextureRegion

    fun dispose()
}

class SimpleTextureProvider(internal var texture: Texture) : TextureProvider {
    internal var renderFrame: TextureRegion

    init {
        this.renderFrame = TextureRegion(texture)
    }

    override fun getFrame(rotationD: Float): TextureRegion {
        return renderFrame
    }

    override fun dispose() {
        texture.dispose()
    }
}
